nca/CoreDS/ConnectivityProperties.py

```diff
@@ -0,0 +1,640 @@
+
+        In the optimized solution, the resulting ConnectivityProperties should not contain named ports:
+            they are substituted with corresponding port numbers, per peer
+        In the original solution, the resulting ConnectivityProperties may contain named ports;
+            they cannot yet be resolved, since dst peers are not provided at this stage the original solution;
+            they will be resolved by convert_named_ports call during query runs.
+
+        :param ConnectivityCube conn_cube: the input connectivity cube including all dimension values,
+            whereas missing dimensions are represented by their default values (representing all possible values).
+        """
+
+        src_ports = conn_cube["src_ports"]
+        dst_ports = conn_cube["dst_ports"]
+        dst_peers = conn_cube["dst_peers"]
+        assert not src_ports.named_ports and not src_ports.excluded_named_ports
+        if (not dst_ports.named_ports and not dst_ports.excluded_named_ports) or not dst_peers:
+            # Should not resolve named ports
+            return ConnectivityProperties.create_props_from_cube(conn_cube)
+
+        # Initialize conn_properties
+        if dst_ports.port_set:
+            dst_ports_no_named_ports = PortSet()
+            dst_ports_no_named_ports.port_set = dst_ports.port_set.copy()
+            conn_cube["dst_ports"] = dst_ports_no_named_ports
+            conn_properties = ConnectivityProperties.create_props_from_cube(conn_cube)
+        else:
+            conn_properties = ConnectivityProperties.make_empty_props()
+
+        # Resolving dst named ports
+        protocols = conn_cube["protocols"]
+        assert dst_peers
+        for peer in dst_peers:
+            real_ports = ConnectivityProperties.resolve_named_ports(dst_ports.named_ports, peer, protocols)
+            if real_ports:
+                conn_cube.update({"dst_ports": real_ports, "dst_peers": PeerSet({peer})})
+                conn_properties |= ConnectivityProperties.create_props_from_cube(conn_cube)
+            excluded_real_ports = ConnectivityProperties.resolve_named_ports(dst_ports.excluded_named_ports, peer, protocols)
+            if excluded_real_ports:
+                conn_cube.update({"dst_ports": excluded_real_ports, "dst_peers": PeerSet({peer})})
+                conn_properties -= ConnectivityProperties.create_props_from_cube(conn_cube)
+        return conn_properties
+
+    @staticmethod
+    def make_empty_props():
+        """
+        Returns empty connectivity properties, representing logical False
+        :return: ConnectivityProperties
+        """
+        return ConnectivityProperties()
+
+    @staticmethod
+    def make_all_props():
+        """
+        Returns all connectivity properties, representing logical True
+        :return: ConnectivityProperties
+        """
+        return ConnectivityProperties(True)
```

Comment:
I added the following comment in the `ConnectivityProperties `class:
The `src_peers` and `dst_peers` dimensions are special dimensions,  they do not have constant domain. Their domain
        depends on the current set of peers in the system (as appears in `BasePeerSet` singleton). This set grows when
        adding more configurations. Thus, there is no unique 'all values' representation. In particular, those
        dimensions are never reduced to inactive.
        This might be a problem in comparison and inclusion operators of `ConnectivityProperties`. The possible solution
        may be to keep 'reference full domain value' for these dimensions (as another member in the `BasePeerSet`),
        and to set it to relevant values per query, and to make a special treatment of these dimensions
        in the above operators.

Reply:
Opened issue https://github.com/IBM/network-config-analyzer/issues/485 about it.